From ef663d710773f87a74a9c1be6b3665462d927dbf Mon Sep 17 00:00:00 2001
From: Gautham Kumar Senthilkumaran <quic_gauthamk@quicinc.com>
Date: Wed, 6 Mar 2024 23:12:40 +0530
Subject: [PATCH] wifi: cfg80211: Adding new condition for ieee80211 hardware register

Bringup support for SLO mode with non-single wiphy architecture.
With non-single wiphy archtecture, all wiphy will hold only one radio interface for each
ath12k radio group.So, existing condition is changed during the ath12k mac hw register.

Signed-off-by: Gautham Kumar Senthilkumaran <quic_gauthamk@quicinc.com>

--- a/net/wireless/core.c
+++ b/net/wireless/core.c
@@ -854,7 +854,7 @@ static int cfg80211_validate_per_hw_chan
          * to advertise channel list for one hw, sband alone should
          * be sufficient
          */
-        if (wiphy->num_hw < 2)
+        if (wiphy->num_hw < 1)
                 return -EINVAL;
 
         for (i = 0; i < wiphy->num_hw; i++) {
